packages/GA4Client/src/commerce-handler.js

```diff
@@ -136,10 +136,147 @@ CommerceHandler.prototype.logCommerceEvent = function (event) {
             null;
     }
 
-    gtag('event', mapGA4EcommerceEventName(event), ga4CommerceEventParameters);
+    this.sendCommerceEventToGA4(
+        mapGA4EcommerceEventName(event),
+        ga4CommerceEventParameters
+    );
     return true;
 };
 
+CommerceHandler.prototype.sendCommerceEventToGA4 = function (
+    eventName,
+    eventAttributes
+) {
+    if (this.common.forwarderSettings.measurementId) {
+        eventAttributes.send_to = this.common.forwarderSettings.measurementId;
+    }
+
+    gtag('event', eventName, eventAttributes);
+};
+
+// Google previously had a CheckoutOption event, and now this has been split into 2 GA4 events - add_shipping_info and add_payment_info
+// Since MP still uses CheckoutOption, we must map this to the 2 events using custom flags.  To prevent any data loss from customers
+// migrating from UA to GA4, we will set a default `set_checkout_option` which matches Firebase's data model.
+CommerceHandler.prototype.logCheckoutOptionEvent = function (
+    event,
+    affiliation
+) {
+    try {
+        var customFlags = event.CustomFlags,
+            GA4CommerceEventType = customFlags[GA4_COMMERCE_EVENT_TYPE],
+            ga4CommerceEventParameters;
```

Comment:
I think we should start splitting up our variable declarations when we migrate code.

```suggestion
        var customFlags = event.CustomFlags;
        var GA4CommerceEventType = customFlags[GA4_COMMERCE_EVENT_TYPE];
        var ga4CommerceEventParameters;
```